internal/handlers/handlers.go

```diff
@@ -91,16 +107,65 @@ func (h *URLHandler) getURL(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	url, exists := h.shortener.Expand(r.Context(), id)
+	url, exists, isDeleted := h.shortener.Expand(r.Context(), id)
 
 	if !exists {
 		http.Error(w, "URL not found", http.StatusBadRequest)
 		return
 	}
 
+	if isDeleted {
+		http.Error(w, "URL has been deleted", http.StatusGone)
+		return
+	}
+
 	http.Redirect(w, r, url, http.StatusTemporaryRedirect)
 }
 
+func (h *URLHandler) GetURLs(w http.ResponseWriter, r *http.Request) {
+	userID, ok := middleware.GetUserIDFromContext(r.Context())
+	if !ok || userID == "" {
+		// Если userID нет в контексте или он пустой - возвращаем 401
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	urls, err := h.shortener.GetUserURLs(r.Context(), userID)
+	if err != nil {
+		logger.Sugar.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if len(urls) == 0 {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	// Преобразуем в нужный формат ответа
+	response := make([]struct {
+		ShortURL    string `json:"short_url"`
+		OriginalURL string `json:"original_url"`
+	}, len(urls))
+
+	for i, url := range urls {
+		response[i] = struct {
+			ShortURL    string `json:"short_url"`
+			OriginalURL string `json:"original_url"`
+		}{
+			ShortURL:    h.config.BaseURL + "/" + url.ShortURL,
+			OriginalURL: url.OriginalURL,
+		}
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		logger.Sugar.Error(err)
```

Comment:
К ошибкам лучше добавлять контекст, чтобы было понятно, что именно пошло не так:

```suggestion
		logger.Sugar.Errorf("failed to encode response: %v", err)
```